lib/Conversion/TritonGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -499,10 +500,7 @@ struct AtomicRMWOpConversion
       // mask
       numElems = tensorTy.getNumElements();
     }
-    Value mask = int_val(1, 1);
-    auto tid = tid_val();
-    mask = and_(mask,
-                icmp_slt(mul(tid, i32_val(elemsPerThread)), i32_val(numElems)));
+    Value mask = getMask(valueTy, rewriter, loc);
```

Comment:
This one adds a cmp eq, so it changes the order of operations, and thus the lit test order changes